### PR TITLE
Fix hack/update-codegen.sh

### PR DESCRIPTION
**Pull request by zshi-redhat**
1. add --go-head-file for missing boilerplate error
2. restore generate-groups.sh file permission
3. update sriovnetwork/v1/zz_generated.deepcopy.go header

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -4,4 +4,6 @@ chmod +x vendor/k8s.io/code-generator/generate-groups.sh
 ./vendor/k8s.io/code-generator/generate-groups.sh all \
       github.com/openshift/sriov-network-operator/pkg/client \
       github.com/openshift/sriov-network-operator/pkg/apis \
-      sriovnetwork:v1
+      sriovnetwork:v1 \
+      --go-header-file hack/boilerplate.go.txt
+chmod -x vendor/k8s.io/code-generator/generate-groups.sh
```
Comment (SchSeba) [inline hack/update-codegen.sh:9]: we already have this line here https://github.com/openshift/sriov-network-operator/pull/270/files#diff-5345a009f642e55fc0d5b16d4f5d6f9bR3

Comment (zshi-redhat) [reply hack/update-codegen.sh:9]: @SchSeba I opened this link, but it didn't switch to a certain line. could you point me to the duplicate code?

Comment (zshi-redhat) [reply hack/update-codegen.sh:9]: For line 9, I remove the execution persmission from the generate-groups.sh file that was previously granted before executing the script. This avoid developer from wrongly submitting the file permission change.

File: hack/boilerplate.go.txt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/*
+Copyright The Kubernetes Authors.
```
Comment (pliurh) [inline hack/boilerplate.go.txt:2]: Do we need this line?

Comment (zshi-redhat) [reply hack/boilerplate.go.txt:2]: This line was included in many pkg/client files already.
I think we can change it if necessary.

**Comment by zshi-redhat** [comment]
@SchSeba @pliurh any further comments on this one?

File: hack/boilerplate.go.txt (hunk 2 above)
Comment (pliurh) [reply hack/boilerplate.go.txt:2]: As the sriov operator is not part of the k8s project but part of openshift project, so I think maybe we should change it to something like `The Openshift Authors` instead.

Comment (zshi-redhat) [reply hack/boilerplate.go.txt:2]: so we will change all the affected files in sriov operator repo?

**Pull request merged**